python/ql/lib/semmle/python/dataflow/new/internal/LocalSources.qll

```diff
@@ -133,6 +137,19 @@ class LocalSourceNode extends Node {
   LocalSourceNode backtrack(TypeBackTracker t2, TypeBackTracker t) { t2 = t.step(result, this) }
 }
 
+/**
+ * A LocalSourceNode that is not a ModuleVariableNode
+ * This class provides a positive formulation of that in its charpred.
+ */
+class LocalSourceNodeNotModule extends LocalSourceNode {
```

Comment:
I could not for the life of me figure out why you needed to know which local source nodes did _not_ refer to a module, when I first saw this class being used. I think we're getting into VeryLongNames territory, but in this case I think the readability is worth the cost!

```suggestion
class LocalSourceNodeNotModuleVariableNode extends LocalSourceNode {
```


I'm also wondering if you really need it. Instead changing `trackUseNode` as

```diff
-      result instanceof DataFlow::ExprNode
+      result instanceof DataFlow::LocalSourceNodeNotModule
```

couldn't we do

```diff
-      result instanceof DataFlow::ExprNode
+      (
+        result instanceof DataFlow::ExprNode
+        or
+        // for captured variables
+        result.asVar() instanceof ScopeEntryDefinition
+      )
```

(I'm not 100% sure that is **better**, but seems like an alternative solution -- let me know what your thoughts are on this)